tests/baselines/reference/controlFlowOptionalChain.errors.txt

```diff
@@ -770,6 +771,8 @@ tests/cases/conformance/controlFlow/controlFlowOptionalChain.ts(567,21): error T
     while (arr[i]?.tag === "left") {
         i += 1;
         if (arr[i]?.tag === "right") {
+            ~~~~~~~~~~~~~~~~~~~~~~~
+!!! error TS2367: This condition will always return 'false' since the types '"left"' and '"right"' have no overlap.

```

Comment:
Looks like it also handle arrays like in: https://github.com/microsoft/TypeScript/issues/10530
but this shouldn't throw an error here - I've proposed a fix for that in my PR:  https://github.com/microsoft/TypeScript/pull/46121

Reply:
@easyrider The https://github.com/microsoft/TypeScript/issues/40617 depends on handling identifies in element access expressions, so some of the cases can be resolved by this PR. `controlFlowOptionalChain.errors.txt` is an outdated test. It was fixed in the latest version.